scipy/spatial/_spherical_voronoi.py

```diff
@@ -32,6 +32,17 @@ def sphere_check(points, radius, center):
     return abs(max_discrepancy)
 
 
+def calculate_solid_angles(R):
+    """Calculates the solid angles of plane triangles. Implements the method of
+    Van Oosterom and Strackee [VanOosterom]_. Assumes that input points have
+    unit norm."""
+    numerator = np.einsum('ij,ij->i', R[:, 0], np.cross(R[:, 1], R[:, 2]))
+    denominator = 1 + np.einsum('ij,ij->i', R[:, 0], R[:, 1])\
+                    + np.einsum('ij,ij->i', R[:, 1], R[:, 2])\
+                    + np.einsum('ij,ij->i', R[:, 2], R[:, 0])
+    return np.abs(2 * np.arctan2(numerator, denominator))
```

Comment:
I think that for all approaches to calculating spherical polygon surface areas there is some kind of weakness related to the accuracy of the spherical excess / angle calculations--here you don't have an interface to modulate the accuracy with i.e., discretization, which is a bit different than I did in #8069, so I'd like us to look into accuracy issues here---I'm happy for your approach to simply be better if it is :)

The authors of the manuscript you cite have clearly thought about this, and do note that there are 32 multiplications, 20 additions, 3 square roots, and 1 ATAN2. They also cite a discussion with a colleague at the University of Amsterdam about numerical stability.